audit.py

```diff
@@ -226,94 +434,114 @@ def repos(
-        empty_columns = [None, None]  # Name, Pushed at
-        if include_archived_repositories:
-            table.add_column("Archived", justify="center")
-            empty_columns.append(None)
-        if include_forked_repositories:
-            table.add_column("Fork", justify="center")
-            empty_columns.append(None)
-        table.add_column("Pushed at")
-        table.add_column("Pull request title")
-        table.add_column("Created at")
-        for repo in sorted(repositories, key=lambda x: x['name'].lower()):
-            repo_row = [repo["name"]]
-            if include_archived_repositories:
-                repo_row.append(format_bool(repo["archived"]))
-            if include_forked_repositories:
-                repo_row.append(format_bool(repo["fork"]))
-            repo_row.append(f'{format_friendly_timestamp(repo["pushed_at"], started)}')
-            if repo["open_pull_requests"]:
-                first_pr = repo["open_pull_requests"][0]
-                repo_row.extend([first_pr["title"], f'{format_friendly_timestamp(first_pr["created_at"], started)}'])
-            table.add_row(*repo_row)
-            for pr in repo["open_pull_requests"][1:]:
-                pr_row = empty_columns + [pr["title"], f'{format_friendly_timestamp(pr["created_at"], started)}']
-                table.add_row(*pr_row)
-        output_text_elements(title, table, started, output)
-
-    else:
+    report_class = {
+        OutputFormat.html: RepoHtmlReport,
+        OutputFormat.text: RepoTextReport,
+    }.get(output_format)
+    if report_class is None:
         OutputFormat.unknown(output_format)
+        return
+
+    report = report_class(started, output, include_archived_repositories, include_forked_repositories)
+    report.begin_report(title)
+    report.begin_table()
+    for repo in sorted(repositories, key=lambda x: x['name'].lower()):
+        open_pull_requests = repo["open_pull_requests"]
+        columns = [
+            repo["name"],
+            repo["archived"],
+            repo["fork"],
+            repo["pushed_at"],
+            open_pull_requests[0]["title"] if open_pull_requests else "",
+            open_pull_requests[0]["created_at"] if open_pull_requests else "",
+        ]
+        report.table_row(*columns, rowspan=len(open_pull_requests), first=True)
+        for pr in open_pull_requests[1:]:
+            report.table_row(title=pr["title"], created_at=pr["created_at"])
+    report.end_table()
+    report.end_report()
+
+
+class RepoContributionsTextReport(TextReportBase):
+
+    def __init__(self, started, output, include_archived_repositories, include_forked_repositories):
+        super().__init__(started, output)
+        self.include_archived_repositories = include_archived_repositories
+        self.include_forked_repositories = include_forked_repositories
+
+    def _table_header(self):
+        if self.table is None:
+            raise RuntimeError("not building a table")
```

Comment:
Can these checks be moved to the super class so the if-statement can be replaced by `super()._table_header()`?

Reply:
Indeed